### PR TITLE
chore(eslint): added eslint-node-plugin

### PR DESCRIPTION
**Pull request by TheLarkInn**
**What kind of change does this PR introduce?**
Refactor, ESLINT ❤️ 

**Did you add tests for your changes?**
None needed!

**If relevant, link to documentation update:**
N/A

**Summary**
Because we have to support node >=4.7 we need to have a easy way to ensure that all linting and node api features are correctly in place. 

This PR adds eslint-node-plugin which allows us to essentially have our `engine` property set, and then get intelligent and appropriate linting settings with the exact "ecmaFeatures" that we are looking for. Aside from that I fixed any lint warnings in master. Now folks will be forced to add "use strict" 

**Does this PR introduce a breaking change?**
No. 

**Other information**

### CODE REVIEW

File: lib/EvalSourceMapDevToolModuleTemplatePlugin.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ EvalSourceMapDevToolModuleTemplatePlugin.prototype.apply = function(moduleTempla
 		}
 		sourceMap.sourceRoot = options.sourceRoot || "";
 		sourceMap.file = module.id + ".js";
-		var footer = self.sourceMapComment.replace(/\[url\]/g, "data:application/json;charset=utf-8;base64," + new Buffer(JSON.stringify(sourceMap)).toString("base64"));
+		var footer = self.sourceMapComment.replace(/\[url\]/g, "data:application/json;charset=utf-8;base64," + Buffer.from(JSON.stringify(sourceMap)).toString("base64"));
```
Comment (TheLarkInn) [inline lib/EvalSourceMapDevToolModuleTemplatePlugin.js]: `new Buffer()` has been deprecated since node 6. So Using Buffer.from()

File: bin/convert-argv.js [hunk 2]
```diff
@@ -466,12 +466,6 @@ module.exports = function(yargs, argv, convertOptions) {
 			options.plugins.push(new ProvidePlugin(name, value));
 		});
 
-		ifBooleanArg("labeled-modules", function() {
```
Comment (bebraw) [inline bin/convert-argv.js]: Why to remove this? Should the removal be a separate PR?

File: lib/Compiler.js [hunk 3]
```diff
@@ -321,7 +321,7 @@ Compiler.prototype.emitAssets = function(compilation, callback) {
 				}
 				var content = source.source();
 				if(!Buffer.isBuffer(content))
```
Comment (bebraw) [inline lib/Compiler.js]: I would force braces with ifs (separate PRs) as the form without can hide errors easily.

File: bin/convert-argv.js (hunk 2 above)
Comment (TheLarkInn) [inline bin/convert-argv.js]: So this is because The file doesn't exist. This was also a lint error.

Comment (bebraw) [inline bin/convert-argv.js]: Ok.

File: lib/Compiler.js (hunk 3 above)
Comment (TheLarkInn) [inline lib/Compiler.js]: Yes thats a good idea

Comment (TheLarkInn) [inline lib/Compiler.js]: I'll do that in a separate PR. Otherwise looks good?

Comment (bebraw) [inline lib/Compiler.js]: Yup.

**Comment by TheLarkInn** [comment]
Rerunning travis to make sure the PR hook was just a flake run. Otherwise maybe related to yarn PR merge.

File: lib/Compiler.js (hunk 3 above)
Comment (Kovensky) [inline lib/Compiler.js]: Maybe `"curly": ["error", "multi-line", "consistent"]`, to still allow for short `if`s that cannot visually be confused with a block?

File: lib/EvalSourceMapDevToolModuleTemplatePlugin.js (hunk 1 above)
Comment (helloyou2012) [inline lib/EvalSourceMapDevToolModuleTemplatePlugin.js]: `Buffer.from` is not existed on `node@v4`

Comment (TheLarkInn) [inline lib/EvalSourceMapDevToolModuleTemplatePlugin.js]: It does infact exist on node 4.7>.

Comment (TheLarkInn) [inline lib/EvalSourceMapDevToolModuleTemplatePlugin.js]: Yup this got reverted!
